### ISSUE TITLE
Annotation targeting by primary key and column names

### ISSUE THREAD

**Issue by noamross** [w3c/csvw#807]
This spec describes cell annotation targeting column and row numbers, but it may also be desirable to annotate data by primary key and column, which would stable when rows are re-arranged or filtered.

Here is how this type of annotation might look:

```
{
  "@context": "http://www.w3.org/ns/csvw",
  "url": "countries.csv",
  "tableSchema": {
  "columns": [{
    "name": "country",
    "titles": "country"
  },{
    "titles": "country group"
  },{
    "titles": "name (en)"
  },{
    "titles": "name (fr)"
  },{
    "titles": "name (de)"
  },{
    "titles": "latitude"
  },{
    "titles": "longitude"
  }],
  "primaryKey": "country"
}
  "notes": [{
    "@type": "oa:Annotation",
    "oa:hasTarget": { "@id": "countries.csv#primaryKey=bg,fr&column=lattitude,longitude" },
    "oa:hasBody": { "oa:text": "These locations are of representative points." },
    "oa:motivatedBy": { "@id": "oa:commenting" }
  }, {
    "@type": "oa:Annotation",
    "oa:hasTarget": { "@id": "countries.csv#primaryKey=bg&column=name (fr)" },
    "oa:hasBody": { "oa:text": "Corrected." },
    "oa:motivatedBy": { "@id": "oa:editing" }
  }]
}
```

Cell ranges would not be applicable in this case.

**Comment by gkellogg** [comment]
I believe what you want to do is set an `@aboutUrl`, which will set the subject URI for each row, which you can then target from your annotation. As `@aboutUrl` takes a URI Template, you can add information from whatever columns you need.

(Also, you may want to set the `language` annotation on columns in different languages, but that's an aside).

**Comment by JeniT** [comment]
@noamross Yes, that could be useful. It would require a different way of defining the fragment identifiers for CSV files (ie a replacement or update to RFC7111).

**Comment by iherman** [comment]
Per the response of @JeniT (https://github.com/w3c/csvw/issues/807#issuecomment-183700505) I am tempted to close this issue; it is indeed not in the charter of this Working Group to change RFC71111...

Any objections?

**Comment by noamross** [comment]
None from me. Thanks for consideration. I'm new to this process; where is
the best place to bring this up for RFC7111?

**Comment by JeniT** [comment]
Any thoughts @dret?

**Comment by dret** [comment]
RFC 7111 is about text/csv. this issue is about something else, it talks about primary keys which are not something that text/csv defines. as such, i think that this issue is unrelated to RFC 7111 and text/csv.

**Comment by GeorgFerdinandSchneider** [comment]
Following the suggestion by @noamross I would like to raise a similar issue.

Coming from relational data bases, the property `csvw:primaryKey` with domain `csvw:Row` seems to be akward. One or more **columns** of a table are typically defined as primary keys of a table.

Despite RFC 7111, it would be great if columns could be annotated to be primary keys or not.

Best

Georg
